### PR TITLE
GEOS-8239 Support Limits and LayerDetails in geofence internal server…

### PR DESCRIPTION
**Pull request by NielsCharlier**
… REST

https://osgeo-org.atlassian.net/browse/GEOS-8239

### CODE REVIEW

**Comment by NielsCharlier** [comment]
Depends on https://github.com/geotools/geotools/pull/1662

File: src/community/geofence-server/src/main/java/org/geoserver/geofence/rest/xml/MultiPolygonAdapter.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
+package org.geoserver.geofence.rest.xml;
```
Comment (bencaradocdavies) [inline src/community/geofence-server/src/main/java/org/geoserver/geofence/rest/xml/MultiPolygonAdapter.java]: @NielsCharlier this file should have a copyright header.

File: src/community/geofence-server/src/test/java/org/geoserver/geofence/rest/MultiPolygonAdapterTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
+package org.geoserver.geofence.rest;
```
Comment (bencaradocdavies) [inline src/community/geofence-server/src/test/java/org/geoserver/geofence/rest/MultiPolygonAdapterTest.java]: @NielsCharlier this file should also have a copyright header (even though it is a test).

**Comment by bencaradocdavies** [comment]
https://github.com/geotools/geotools/pull/1662 is merged so I will close/open this PR to retrigger Travis when the GeoTools build has completed and new artifacts are available.

This PR looks clean (except for the two missing copyright headers) but I know nothing about geofence. Could someone like @etj take a quick look? Who else knows geofence?

**Comment by bencaradocdavies** [comment]
Close/open to retrigger Travis.

**Comment by bencaradocdavies** [comment]
@NielsCharlier, if you want the GeoTools build to deploy the gt-xsd-gml3 tests jars so that you can use them as a dependency in GeoServer, copy the maven-source-plugin and maven-jar-plugin stanzas from the gt-app-schema pom to the gt-xsd-gml3 pom.

**Comment by tbarsballe** [comment]
@bencaradocdavies - just a comment, instead of closing and reopening to retrigger a build, you can click the "Details" link beside the Travis Build, log into travis using you github account, and then you will have access to a button that restarts the build.

**Comment by bencaradocdavies** [comment]
@tbarsballe unfortunately the "Restart build" button in Travis reruns the build with exactly the same Travis settings as the original build. AFAIK the only way to restart a build with Travis settings that have been updated on the target branch is to close and then reopen. While I agree that the "Restart build" button in Travis is the usual and preferred way to restart a build, this PR predates my Travis settings changes on master and needed the new settings. Travis needed `MAVEN_OPTS=-Xmx512m` to avoid getting killed for exceeding the 4 GB container memory limit; the default settings allowed the heap to grow to over 1.6 GB, and with four 512 MB test threads, we often had nondeterministic build failures.

**Comment by tbarsballe** [comment]
@bencaradocdavies Ah, I didn't know that. Thanks for the info, good to know.

**Comment by NielsCharlier** [comment]
Thanks @bencaradocdavies, I didn't know what was going on.

**Comment by bencaradocdavies** [comment]
https://github.com/geotools/geotools/pull/1685 is merged. Once the artifacts are deployed, the Travis build for this PR can be restarted.

**Comment by bencaradocdavies** [comment]
@NielsCharlier, the GeoTools build has completed and, now that the gt-xsd-gml3 tests jar is deployed, I have restarted the Travis build for this PR.

**Comment by bencaradocdavies** [comment]
@NielsCharlier the Travis build has passed.
* This PR should be reviewed by someone with some knowledge of geofence (i.e. not me). I have suggested @etj but we have not have a response. Can you think of anyone else? Does this module have a maintainer?
* The code looks clean and has decent unit test coverage.
* Two source code files lack copyright headers.
* There are no comments or javadoc added anywhere in this PR, despite the addition of several new classes and public methods. The code is clean and readable and method names are close to self-explanatory, but class javadocs should be added at the least. Yes, this is a community module, but better now than retrofitting docs at graduation time.
* The release of 2.12-beta is on Sunday, that is, very soon. If not merged by then, this new feature will be blocked by the feature freeze. If you intend to include this feature in 2.12-beta, please address all the issues above and make yourself known to @jodygarnett on this PR and by replying to his feature freeze email on geoserver-devel.

**Comment by NielsCharlier** [comment]
@bencaradocdavies Thanks again, I have added headers and some basic javadoc. I guess it's stuck in the FF now.  I wrote this entire module so I guess I'm as close to a maintainer as it gets.
Will wait for review from @etj, or @afabiani would be an alternative reviewer.

**Comment by bencaradocdavies** [comment]
@NielsCharlier if you are the de facto maintainer of this community module, then review, while nice to have, is not mandatory. I am tempted to just merge it. As far as I can tell, 2.12-beta has not yet been released, so the freeze is not yet in effect.

Merging this PR is at the discretion of release manager @jodygarnett. Jody can this PR be merged on master?

**Comment by bencaradocdavies** [comment]
@aaime advises that community modules are not part of the release and thus not affected by the code freeze, so I have merged.

File: src/community/geofence-server/src/main/resources/applicationContext.xml [hunk 3]
```diff
@@ -155,32 +155,10 @@
...
-                <!-- XML view using a JAXB marshaller -->
-                <bean id="jaxbView" class="org.springframework.web.servlet.view.xml.MarshallingView">
-                    <constructor-arg>
-                        <bean id="jaxb2Marshaller" class="org.springframework.oxm.jaxb.Jaxb2Marshaller">
-                            <property name="classesToBeBound">
-                                <list>
-                                    <value>org.geoserver.geofence.rest.xml.JaxbRule</value>
-                                    <value>org.geoserver.geofence.rest.xml.JaxbRuleList</value>
+
                                     <value>org.geoserver.geofence.rest.xml.JaxbRoleList</value>
```
Comment (bencaradocdavies) [inline src/community/geofence-server/src/main/resources/applicationContext.xml]: @NielsCharlier one other question (after I merged the PR!): these lines look out of place and are likely ignored. You have removed the enclosing bean.

Comment (NielsCharlier) [reply src/community/geofence-server/src/main/resources/applicationContext.xml]: You're right, that is odd. Bad merge. Fixing.

**Pull request merged**